_posts/2016-05-16-rust-at-one-year.md

```diff
@@ -0,0 +1,269 @@
+  unique HTML5 parser that leverages Rust's macro system to make the code
+  resemble the spec as closely as possible. These are just scratching the
+  surface, of course, and ecosystem growth, curation and
+  coherence&mdash;particularly around async IO and the web stack&mdash;will
+  continue to be a major focus in the coming year.
+
+[filesystem access]: http://static.rust-lang.org/doc/master/std/fs/index.html
+[networking]: http://static.rust-lang.org/doc/master/std/net/index.html
+[time]: http://static.rust-lang.org/doc/master/std/time/index.html
+[collections]: http://static.rust-lang.org/doc/master/std/collections/index.html
+[libc]: https://github.com/rust-lang/libc
+[winapi]: https://github.com/retep998/winapi-rs
+[gcc]: https://github.com/alexcrichton/gcc-rs
+[mio]: https://github.com/carllerche/mio/
+[rayon]: http://smallcultfollowing.com/babysteps/blog/2015/12/18/rayon-data-parallelism-in-rust/
+[crossbeam]: http://aturon.github.io/blog/2015/08/27/epoch/
+[regex]: https://github.com/rust-lang-nursery/regex
+[diesel]: http://diesel.rs/
+[nom]: https://github.com/Geal/nom
+[lalrpop]: http://smallcultfollowing.com/babysteps/blog/2015/09/14/lalrpop/
+[html5ever]: https://kmcallister.github.io/talks/rust/2014-rust-macros/slides.html
+
+- **Platforms and targets**. Rust's footprint is not much bigger than C's, which
+  makes it ideal for using in all kinds of places. Over the last year, Rust
+  gained the ability to work directly with the native [MSVC toolchain] on
+  Windows, to target [musl] (thereby creating a binary that can be used with
+  zero dependencies on *any* variety of Linux), to target Android and ARM
+  devices, and [many more platforms][platforms]. The new [rustup tool] makes it
+  a breeze to manage and compile to these various targets. As of
+  [Rust 1.6][no_std], you can use Rust without its full standard library,
+  limiting to a core library that does not require any OS services (and hence is
+  suitable for [writing OSes](http://os.phil-opp.com/)
+  [in Rust][intermezzos]). Finally, there are an increasing number of libraries
+  for embedding Rust code into other contexts, like [node.js][neon],
+  [Ruby][helix] and [Go][rure-go].
+
+[MSVC toolchain]: https://github.com/rust-lang/rust/pull/25350
+[MUSL]: https://www.musl-libc.org/
+[platforms]: https://forge.rust-lang.org/platform-support.html
+[rustup tool]: http://blog.rust-lang.org/2016/05/13/rustup.html
+[no_std]: http://blog.rust-lang.org/2016/01/21/Rust-1.6.html
+[intermezzos]: https://intermezzos.github.io/
+[neon]: http://calculist.org/blog/2015/12/23/neon-node-rust/
+[helix]: http://blog.skylight.io/introducing-helix/
+[rure-go]: https://github.com/BurntSushi/rure-go
+
+- **Tools**. Because Rust looks just like C on the outside, it's instantly
+  usable with a wide range of existing tools; it works out of the box with
+  [lldb], [gdb], [perf], [valgrind], [callgrind], and many, many more.  Our
```

Comment:
it's probably fine to say that this is true **in principle**, and that the community has made great strides making it usually true **in practice**.